samples/project-with-buildSrc/buildSrc/src/main/kotlin/MyProjectExtension.kt

```diff
@@ -17,8 +17,8 @@ import org.gradle.api.model.ObjectFactory
 
 import org.gradle.kotlin.dsl.*
 
-
-open class MyProjectExtension(objects: ObjectFactory) {
+@my.AllOpen
+class MyProjectExtension(objects: ObjectFactory) {
```

Comment:
Is this really any better?

Reply:
Not really, I would say both are bad!
The problem is that there are not common interface for Project extensions AFAIK. 

A solution would be to add to an empty interface 

```kt
// MyProjectExtension.kt
class MyProjectExtension(objects: ObjectFactory) : my.Extension

// my/allopen-compiler-plugin.kt
package my

@AllOpen 
interface Extension
```

Reply:
I went forward and did that